R/blastula_email_format.R

```diff
@@ -0,0 +1,121 @@
+#' @param md_extensions Markdown extensions to be added or removed from the
+#'   default definition or R Markdown.
+#' @param connect_footer Should a prepared footer message with links be included
+#'   in the rendered email?
+#' @param ... Specify other options in [rmarkdown::html_document()].
+#'
+#' @export
+blastula_email <- function(toc = FALSE,
+                           toc_depth = 3,
+                           toc_float = FALSE,
+                           number_sections = FALSE,
+                           section_divs = TRUE,
+                           fig_width = 5.35,
+                           fig_height = 5,
+                           fig_retina = 2,
+                           fig_caption = TRUE,
+                           dev = "png",
+                           smart = TRUE,
+                           self_contained = TRUE,
+                           template = "blastula",
+                           includes = NULL,
+                           keep_md = FALSE,
+                           md_extensions = NULL,
+                           connect_footer = FALSE,
+                           ...) {
+
+  if (template == "blastula") {
+    template <- system.file("rmd", "template.html", package = "blastula")
+  }
+
+  pandoc_args <- NULL
+  if (isTRUE(connect_footer)) {
+    pandoc_args <- c(
+      pandoc_args,
+      "--variable=rsc-footer:1",
+      # Note bash escaping is handled by R Markdown (via shQuote())
+      paste0(
+        "--variable=rsc-date-time:", add_readable_time(time = Sys.time())),
+      paste0(
+        "--variable=rsc-report-url:",
+        Sys.getenv("RSC_REPORT_URL", unset = "https://connect.example.com/content/1234/")),
+      paste0(
+        "--variable=rsc-report-rendering-url:",
+        Sys.getenv("RSC_REPORT_RENDERING_URL", unset = "https://connect.example.com/content/1234/_rev5678")),
+      paste0(
+        "--variable=rsc-report-subscription-url:",
+        Sys.getenv("RSC_REPORT_SUBSCRIPTION_URL", unset = "https://connect.example.com/connect/#/apps/1234/subscriptions")),
+      paste0("--variable=rsc-report-name:", Sys.getenv("RSC_REPORT_NAME"))
+    )
+  }
+
+  rmarkdown::html_document(
+    toc = toc,
+    toc_depth = toc_depth,
+    toc_float = toc_float,
+    number_sections = number_sections,
+    section_divs = section_divs,
+    fig_width = fig_width,
+    fig_height = fig_height,
+    fig_retina = fig_retina,
+    fig_caption = fig_caption,
+    dev = dev,
+    df_print = "default",
+    code_folding = "none",
+    code_download = FALSE,
+    smart = smart,
+    self_contained = self_contained,
+    theme = "default",
+    highlight = "default",
+    mathjax = NULL,
+    template = template,
+    extra_dependencies = NULL,
+    css = NULL,
+    includes = includes,
+    keep_md = keep_md,
+    lib_dir = NULL,
+    md_extensions = md_extensions,
+    pandoc_args = pandoc_args,
+    ...
+  )
+}
+
+bash_escape <- function(x) {
```

Comment:
This (function that I wrote) isn't needed and isn't right, please remove

Reply:
Thanks, it's been removed.